### ISSUE TITLE
incorrect path search in find_graphviz()

### ISSUE THREAD

**Issue by GoogleCodeExporter** [erocarrera/pydot#2]
What steps will reproduce the problem?
1. Install (multiple versions of) graphviz into multiple locations
2. Include in PATH more than one directory containing graphviz
3. call pydot.find_graphviz()

What is the expected output? What do you see instead?

I expect the first executable in the path. ` pydot.find_graphviz()` finds the
last executable in the path.

What version of the product are you using? On what operating system?

`pydot-0.9.10, python-2.5, Ubuntu 7.04 (Intel 32-bit)`

Please provide any additional information below.
Reordering the loops and inserting breaks ought to produce correct results:

```python
        for prg in progs.keys():
                for path in os.environ['PATH'].split(os.pathsep):
                        if os.path.exists(path+os.path.sep+prg):
                                progs[prg] = path+os.path.sep+prg
                                break
                        elif os.path.exists(path+os.path.sep+prg + '.exe'):
                                progs[prg] = path+os.path.sep+prg + '.exe'
                                break
```

Original issue reported on code.google.com by `Li.Any...@gmail.com` on 18 Sep 2007 at 10:37

**Comment by johnyf** [comment]
This suggests another reason for moving registry and custom non-`$PATH`-related heuristics *after* checking the `$PATH`. The first occurrence of GraphViz in the `$PATH` should be preferred in any OS over any other place where GraphViz may be searched for by `pydot`.

**Comment by johnyf** [comment]
Addressed by f3abbd537ed8adfeb2c5562f6c62403094111aa8, which uses the [`subprocess` module](https://docs.python.org/2/library/subprocess.html) from Python's standard library. The error reported in the OP corroborates the decision to remove custom path-searching code, hard-coded paths, and other unexpected and insecure code.

cf. also:
https://github.com/erocarrera/pydot/issues/91#issuecomment-229639358
https://github.com/erocarrera/pydot/issues/109#issuecomment-227346878
